File: Segundo/Programacion Orientada a Objetos/ruby p4/PowerEfficientSpaceStation.rb
require_relative'SpaceStation'
require_relative'Transformation'

module Deepspace
    class PowerEfficientSpaceStation < SpaceStation
        EFFICIENCYFACTOR=1.10
        def fire
            return super*EFFICIENCYFACTOR
        end

        def protection
            return super*EFFICIENCYFACTOR
        end

        def setLoot(loot)
            super
            return Transformation::NOTRANSFORM
        end
    end
end

File: Segundo/PDOO/deepSpace/src/deepspace/PowerEfficientSpaceStation.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author carmenc
 */
public class PowerEfficientSpaceStation extends SpaceStation{
    private static final float EFFICIENCYFACTOR=1.10f;
    
    public PowerEfficientSpaceStation(SpaceStation station){
        super(station);
    }
    
    @Override
    public float fire(){
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override
    public float protection(){
        return super.protection()*EFFICIENCYFACTOR;
    }
    
    @Override
    public Transformation setLoot(Loot loot){
        return Transformation.NOTRANSFORM;
    }
    
    @Override
    public SpaceStationToUI getUIversion(){
        return new PowerEfficientSpaceStationToUI(this);
    }
}
